lib/Job/Job.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Resque\Job;
+
+use Resque\JobHandler;
+
+/**
+ * Base Resque Job class.
+ *
+ * @package		Resque\Job
+ * @author		Heinz Wiesinger <pprkut@liwjatan.org>
+ * @license		http://www.opensource.org/licenses/mit-license.php
+ */
+abstract class Job
+{
+	/**
+	 * Job arguments
+	 * @var array
+	 */
+	public $args;
+
+	/**
+	 * Associated JobHandler instance
+	 * @var JobHandler
+	 */
+	public $job;
+
+	/**
+	 * Name of the queue the job was in
+	 * @var string
+	 */
+	public $queue;
+
+	/**
+	 * (Optional) Job setup
+	 *
+	 * @return void
+	 */
+	public function setUp(): void
+	{
+		// no-op
+	}
+
+	/**
+	 * (Optional) Job teardown
+	 *
+	 * @return void
+	 */
+	public function tearDown(): void
+	{
+		// no-op
+	}
+
+	/**
+	 * Main method of the Job
+	 *
+	 * @return void
+	 */
+	abstract public function perform(): void;
```

Comment:
Unfortunately, this will break any implementations that use the return value to indicate job results in the status keys in Redis. For backward compatibility, and for jobs to be able to use this feature, we should set this to `object|array|string|int|float|bool|null` or `mixed` instead, though either would drop 7.x support.

Reply:
`mixed` also doesn't work, because it breaks all implementations that choose not to return at all. So it would need to allow `void` too.

Reply:
Given that we internally treat it as returning a value, I'd say returning `void` is already incorrect.

Fortunately, this is already a breaking change (mandatory base class instead of optional interface), so we can take some liberties, here.

Reply:
For now, I reverted to the previous signature, without a return type specified. Added tests for this use case too so it doesn't get overlooked in the future.

Reply:
> Given that we internally treat it as returning a value, I'd say returning void is already incorrect.

And yet almost all the test job classes in the unit tests are not returning anything :upside_down_face:

Reply:
Yeah, coding practices have changed quite a bit since this was first written. We'll get it cleaned up as we migrate to PHP 8+. I'd also like to eventually update everything to use the same in-Redis formats as the Ruby implementation, since I'm sure there's been some drift over the years. But that can wait for the moment.